churaverse-plugins-server/src/bombPlugin/bombPlugin.ts

```diff
@@ -46,8 +46,15 @@ export class BombPlugin extends BasePlugin<IMainScene> {
   }
 
   private update(ev: UpdateEvent): void {
-    sendExplodedBomb(this.networkPluginStore.messageSender, this.bombPluginStore.bombs)
-    removeExplodedBomb(this.bombPluginStore.bombs)
+    const explodedBombIds: string[] = this.bombPluginStore.bombs.getAllId().filter((bombId) => {
+      const bomb = this.bombPluginStore.bombs.get(bombId)
+      return bomb?.isExplode === true
+    })
+
+    if (explodedBombIds.length > 0) {
+      removeExplodedBomb(this.bombPluginStore.bombs, explodedBombIds)
+      sendExplodedBomb(this.networkPluginStore.messageSender, explodedBombIds)
```

Comment:
フロントに爆発を通知して、repositoryから削除する方が自然な気がしますが、順番逆だと不都合ありますか？

```suggestion
      sendExplodedBomb(this.networkPluginStore.messageSender, explodedBombIds)
      removeExplodedBomb(this.bombPluginStore.bombs, explodedBombIds)
```

Reply:
確かに、爆発通知→削除の順番が自然なので修正しました